### ISSUE TITLE
Bug: block cursor shown instead of normal one

### ISSUE THREAD

**Issue by thorn0** [facebook/lexical#5786]
Lexical version: 0.14.2

## Steps To Reproduce

1. Open https://stackblitz.com/edit/facebook-lexical-4cu2rd?file=src%2FApp.tsx and notice that the cursor is at the end of the collapsible container, after the horizontal rule. It's correctly shown as a block cursor.
2. Using the right arrow key, start moving the cursor forward.

## The current behavior

The cursor keeps being shown as a block cursor inside the container even though the selection moves to the next paragraph as expected (which can be seen in the debug tree view). The real cursor is invisible. It gets visible when the user starts typing.

## The expected behavior

When the selection moves to the next paragraph, the real cursor should become visible and be shown there, the block cursor should hide.

**Comment by ivailop7** [comment]
Yeah, I think this was related to any element node being the last node. In this case since collapsible container is a shadowRoot, it's essentially the last node inside.

**Comment by etrepum** [comment]
This is still reproducible, whatever heuristic is being used to remove the block cursor just isn't getting triggered when it gets stuck in this state. I think it has something to do with updates that don't go through a full reconciliation.

**Comment by levensta** [comment]
Does not reproduce in version 0.45.0

https://github.com/user-attachments/assets/796fe5f0-8bf4-4fe4-a18d-a3f45fb8d2a0

https://playground.lexical.dev/#doc=H4sIAAAAAAAAE81UTWvDMAz9K0NnryRszTpfB2WHHQY9jh28WGsNihUUpawr-e_DaUub7rDPQk-2JPSk957xGtAHZZmpUwS7BmHWdJaLQF4wgn06CjyqCwQ2M_DKUjntrxV7BAsxZQgMNLqilAADim8KFu6RiFO4qlOhzxpYojSBI9i8ezbgg2CpfQykAvsRCShEj3EzbgtSO3FzcfVigLQZOd0vl8LZbqHOnILdHUd1IaJ8n2Fsib4iWDKRq5vwQnipQQmHgCehMuVzsWnbvmAJ70lfkvZYgb9JWnLUVP5nyN1DOOTKdbJFpcVPrv3WqAcWrC5C3bTVORj2c936z2awa2eAXKMzt0QPNr-Z5MW4yCZXt3lhoOFWytT3SG41F26jP-yGbHQ9HmXQfQB4XxNj1QQAAA
